master/id_allocator.go

```diff
@@ -216,21 +203,20 @@ errHandler:
 func (alloc *IDAllocator) allocateClientID() (clientID uint64, err error) {
 	alloc.mpIDLock.Lock()
 	defer alloc.mpIDLock.Unlock()
-	var cmd []byte
-	metadata := new(RaftCmd)
-	metadata.Op = opSyncAllocClientID
-	metadata.K = maxClientIDKey
-	clientID = atomic.LoadUint64(&alloc.clientID) + 1
-	value := strconv.FormatUint(uint64(clientID), 10)
-	metadata.V = []byte(value)
-	cmd, err = metadata.Marshal()
-	if err != nil {
+	var leader uint64
+	var term uint64
+	leader, term = alloc.partition.LeaderTerm()
+	if leader != alloc.partition.Status().NodeID {
 		goto errHandler
 	}
-	if _, err = alloc.partition.Submit(cmd); err != nil {
+	if alloc.clientCounter == math.MaxUint32 {
+		// FIXME: give up current term
```

Comment:
Master-slave switching occurs during network jitter. In ideal network conditions, master-slave switching occurs rarely. However, if there are 500 ID requests per second for several months, it will overflow after more than three months of continuous operation, putting the risk of client ID conflicts.

Reply:
If there is a possibility of conflicting ids(the allocated id reached the max value of `uint32` in a term), the master will return an error.

This is a temporary scheme.

If we can give up current term and switch to next term, this problem could be solved and the FIXME should be removed.

Reply:
Also see special note for reviewers.

Reply:
> Also see special note for reviewers.

We can add a check process after get the id and retry if exist,the confict is a rare accident

Reply:
Good idea!

I will update this pull request tonight.

Reply:
OK, this feature has been implemented.